parser/src/main/kotlin/com/linkedin/dex/parser/DecodedValue.kt

```diff
@@ -45,20 +52,19 @@ sealed class DecodedValue {
                 is EncodedValue.EncodedFloat -> return DecodedFloat(encodedValue.value)
                 is EncodedValue.EncodedDouble -> return DecodedDouble(encodedValue.value)
                 is EncodedValue.EncodedString -> {
-                    dexFile.byteBuffer.position(dexFile.stringIds[encodedValue.value].stringDataOff)
-                    // read past unused size item
-                    Leb128.readUnsignedLeb128(dexFile.byteBuffer)
-                    return DecodedString(ParseUtils.parseStringBytes(dexFile.byteBuffer))
+                    val position = dexFile.stringIds[encodedValue.value].stringDataOff
+                    return DecodedString(readStringInPosition(dexFile, position))
                 }
                 is EncodedValue.EncodedType -> {
-                    dexFile.byteBuffer.position(dexFile.typeIds[encodedValue.value].descriptorIdx)
-                    // read past unused size item
-                    Leb128.readUnsignedLeb128(dexFile.byteBuffer)
-                    return DecodedType(ParseUtils.parseStringBytes(dexFile.byteBuffer))
+                    val position = dexFile.typeIds[encodedValue.value].descriptorIdx
+                    return DecodedType(readStringInPosition(dexFile, position))
                 }
                 is EncodedValue.EncodedBoolean -> return DecodedBoolean(encodedValue.value)
                 is EncodedValue.EncodedNull -> return DecodedNull
-
+                is EncodedValue.EncodedEnum -> {
+                    val position = dexFile.stringIds[dexFile.fieldIds[encodedValue.value].nameIdx].stringDataOff
```

Comment:
nitpick: I'd suggest to have a `val nameIdx = dexFile.fieldIds[encodedValue.value].nameIdx` to make it easier to read and debug if needed

Reply:
Fixed